### ISSUE TITLE
Have swarm create actions autonomously when action is not available. Have it make github actions, file actions, web scraping actions etc.

### ISSUE THREAD

**Issue by BrianP8701** [BrianP8701/swarmstar#4]
first signs of self sufficiency

**Comment by J3ykob** [comment]
Hey, sometime ago I came up with such a framework for making decisions about actions in the swarm. I really liked your project so will drop it here, maybe will become useful one day:))

## Conversational and decision making framework

Conversational framework for autonomous agents swarm could consist of three rounds.  Each round would constitute the final decision made by the decision board but can also be used for lower level agents.

### First round:
In the first round given the goal each agent (discussion participant) would provide an initial response based on their character’s individual objectives and motivations. Importantly the agents would not have any insight into the answers of the other agents.

### Second round:
Agents would provide another response Now with an insight to all the initial responses given by the agents in the first round.

### Third round:
The agents would respond asynchronously in a predefined way (JSON or sth like that), indicating what action, according to them, should be taken considering all the responses from previous rounds (their decision could be something like: spawn new manager, start the task over, anything honestly). Each response would be then run through an embedding Network, returning a vector, which then would be compared with vector space encodings of available functions in the system. 

This way, the board could effectively decide on the action it wants to take. For example majority Voting or some other Democratic metric could be used here (we just check how many answers from the last round lie within close proximity to an available function in the vector space). If the vector space gave an inconclusive decision, then other agent (function creator agent) could work on creating a new function that would satisfy the majority’s decision

**Comment by BrianP8701** [comment]
Yeah! This could be really helpful for certain decisions within the swarm.

Particularly where we need multiple perspectives, that can't all be squeezed into one context window (filling up the context window reduces performance, and anyway we want each agent to have a specific viewpoint to focus on). So multiple agents focusing on different aspects of the problem.

Example in software automation:
Having separate agents focused on the overarching goal of the project, design/architecture of the code being written, specific task being done now, and documentation of the swarm+current project and anything else. It's essential when writing software to understand the full scope of requirements for that piece of code and how it all fits in. Your method will definitely be considered! Thanks a lot!
